modP2pImpl/src/org/aion/p2p/impl1/P2pMgr.java

```diff
@@ -333,6 +334,7 @@ public void closeSocket(SocketChannel _sc, String _reason, Exception e) {
             SelectionKey sk = _sc.keyFor(selector);
             if (sk != null) {
                 sk.cancel();
+                sk.attach(null);
```

Comment:
is this done to improve garbage collection by reducing reference counting to the previous object?

Reply:
just see some guidance recommend to clean the buffer in case the gc is not smart.